Encryption.md

```diff
@@ -19,216 +19,544 @@
+   know the AAD prefixes are "employees_05_23_18.partition0" etc.
+   * The aad_metadata field contains the AAD prefix itself. In this case, the reader must be able 
+   to verify its validity. For example, if the AAD prefix stored in aad_metadata is "employees_23May2018.part0001", 
+   the reader should know it is fine, but if the aad_metadata stores "employees_23May2016.part0001" - 
+   the version is wrong.
+   * The AAD_Prefix is generated as a random string. Together with a file name, it is sent by the file 
+   writer to a trusted metadata service, deployed in the organization. The reader, that needs to parse a 
+   certain version of a table, authenticates the service and gets the list of table files and their AAD prefixes.
+   * Frameworks above Parquet create table integrity tools that manage the file AAD prefixes. For example, 
+   a framework table writer can create a signature file that contains the table ID, file list/number and 
+   optional AAD prefixes. The AAD prefixes can be either explicit (eg random) or derived from 
+   the table ID and file name/number. The signature file is signed with a secret key (private or symmetric), and 
+   stored next to the table files, or in another location known to the readers. The 
+   readers get the public or symmetric key, verify the signature file and get the AAD prefixes for the table files 
+   (potentially, along with file names/paths).
+   
+#### 4.4.2 AAD_Suffix
+File integrity can be disrupted by module swapping inside the file - such as replacement 
+of column pages in row group 0 by pages from the same column in row group 1. Also, if an encryption 
+key is re-used in multiple files and the user has not provided a unique AAD_Prefix for each file – an 
+attacker can swap modules between the files. While these attacks succeed under rare conditions and in 
+any case don't harm data confidentiality - they can be prevented by using an internally generated file 
+and module identity as a part of GCM AAD formation in Parquet. 
+
+The AAD_Suffix is built by direct concatenation of the following parts: 
+1.	[All modules] random byte array (file identifier): variable length
+2.	[All modules] module type: 1 byte
+3.	[All modules except footer] row group ordinal: 2 byte short (little endian)
+4.	[All modules except footer] column ordinal: 2 byte short (little endian)
+5.	[Data page and header only]: page ordinal: 2 byte short (little endian)
+
+The following module types are defined:  
+
+   * Footer (0)
+   * ColumnMetaData (1)
+   * Data Page (2)
+   * Dictionary Page (3)
+   * Data PageHeader (4)
+   * Dictionary PageHeader (5)
+   * ColumnIndex (6)
+   * OffsetIndex (7)
+
+
+|                      |File ID     | Module type | Row group ordinal | Column ordinal | Page ordinal|
+|----------------------|------------|-------------|-------------------|----------------|-------------|
+| Footer               |     V      |    V (0)    |         X         |       X        |      X      |
```

Comment:
What do `V` and `X` indicate?

Reply:
I would probably use a character to indicate it is used and blank to indicate it is not. `V` for a checkmark isn't clear.

Whatever characters you decide to use, the convention should be explained somewhere. Readers should never guess what something means in a spec.